app/import/robot/police.js

```diff
@@ -1,17 +1,51 @@
 const wreck = require('@hapi/wreck')
 const config = require('../../config/index')
+const { getPostcodeLongLat } = require('./postcode')
+const { getPoliceForces } = require('../../repos/police-forces')
 
-const lookupPoliceForceByPostcode = async (postcode, swallowError) => {
-  const url = `${config.robotImportPoliceApiUrl}?postcode=${postcode}`
-  try {
-    const { payload } = await wreck.get(url, { json: true })
-    return payload
-  } catch (e) {
-    if (!swallowError) {
-      console.log(e)
+const policeBaseUrl = config.policeApi.baseUrl
+const policeLocationEndpoint = 'locate-neighbourhood'
+
+const policeApiOptions = {
+  json: true
+}
+
+const lookupPoliceForceByPostcode = async postcode => {
+  const coords = await getPostcodeLongLat(postcode)
+  const policeForceName = await getPoliceForce(coords)
+  return await matchPoliceForceByName(policeForceName)
+}
+
+const getPoliceForce = async coords => {
+  let retries = 0
+  while (retries < 3) {
+    try {
+      const { payload } = await wreck.get(`${policeBaseUrl}/${policeLocationEndpoint}?q=${coords.lat},${coords.lng}`, policeApiOptions)
+      // Only grab first result, even if many
+      return payload?.force
+    } catch (e) {
+      if (e?.output?.statusCode === 429) {
+        // Rate limiting
```

Comment:
wondering if there's any way to test this (could be unit or otherwise)?  we ran into production issues on last project because some code like this was not properly tested.

Reply:
defo needs coverage